src/core/overflowsafe_type.hpp

```diff
@@ -162,19 +162,11 @@ class OverflowSafeInt
 
 	/* Operators for (in)equality when comparing overflow safe ints. */
 	inline constexpr bool operator == (const OverflowSafeInt& other) const { return this->m_value == other.m_value; }
-	inline constexpr bool operator != (const OverflowSafeInt& other) const { return !(*this == other); }
-	inline constexpr bool operator >  (const OverflowSafeInt& other) const { return this->m_value > other.m_value; }
-	inline constexpr bool operator >= (const OverflowSafeInt& other) const { return this->m_value >= other.m_value; }
-	inline constexpr bool operator <  (const OverflowSafeInt& other) const { return !(*this >= other); }
-	inline constexpr bool operator <= (const OverflowSafeInt& other) const { return !(*this > other); }
+	inline constexpr auto operator <=>(const OverflowSafeInt& other) const { return this->m_value <=> other.m_value; }
 
 	/* Operators for (in)equality when comparing non-overflow safe ints. */
 	inline constexpr bool operator == (const int other) const { return this->m_value == other; }
-	inline constexpr bool operator != (const int other) const { return !(*this == other); }
-	inline constexpr bool operator >  (const int other) const { return this->m_value > other; }
-	inline constexpr bool operator >= (const int other) const { return this->m_value >= other; }
-	inline constexpr bool operator <  (const int other) const { return !(*this >= other); }
-	inline constexpr bool operator <= (const int other) const { return !(*this > other); }
+	inline constexpr auto operator <=>(const int other) const { return this->m_value <=> other; }
```

Comment:
I think we don't need the `operator ==` either here?

Reply:
If you `= default` the `operator<=>`, then you don't need it. If you implement `operator<=>` yourself, then no `operator==` nor `operator!=` are being generated.

I have had a lot of fights over the last few days with my compiler why changing to `operator<=>` was throwing all kinds of ambiguity errors. What was I doing wrong with the implementation of `operator<=>`, until I found a text that stated exactly that what is in the first paragraph.

Reply:
Ah, and you said as much in at the start, sorry!